src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1068,7 +1068,13 @@ bool clang_c_convertert::get_type(const clang::Type &the_type, typet &new_type)
 
   case clang::Type::Enum:
   {
-    new_type = enum_type();
+    const clang::EnumType &ent = static_cast<const clang::EnumType &>(the_type);
+
+    clang::QualType q_type = ent.getDecl()->getPromotionType();
```

Comment:
Shouldn't we use [desugar()](https://clang.llvm.org/doxygen/classclang_1_1EnumType.html#abfac317ad4308d7c91d32c978baf09ea) instead?

Reply:
I tried, but there was a seg fault when I built.

Reply:
What's the cause of the segfault?

Reply:
This error is not clear to me.

<details>

```
[0/2] Re-checking globbed directories...
[4/21] Linking CXX executable src/c2goto/c2goto
/usr/bin/ld: /home/lxzy/ESBMC_Project/clang+llvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04/lib/libLLVMSupport.a(Path.cpp.o): in function `llvm::sys::fs::expandTildeExpr(llvm::SmallVectorImpl<char>&)':
Path.cpp:(.text._ZN4llvm3sys2fsL15expandTildeExprERNS_15SmallVectorImplIcEE+0x12a): warning: Using 'getpwnam' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /home/lxzy/ESBMC_Project/clang+llvm-11.0.0-x86_64-linux-gnu-ubuntu-20.04/lib/libLLVMSupport.a(Path.cpp.o): in function `llvm::sys::path::native(llvm::SmallVectorImpl<char>&, llvm::sys::path::Style)':
Path.cpp:(.text._ZN4llvm3sys4path6nativeERNS_15SmallVectorImplIcEENS1_5StyleE+0x5bc): warning: Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
[5/21] Generating libc model /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64_fp.goto
FAILED: src/c2goto/clib64_fp.goto /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64_fp.goto
cd /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto && /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/c2goto /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/builtin_libs.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/ctype.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/fenv.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/getopt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/inet.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/io.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/kernel.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/semaphore_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/stdlib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/string.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/syserr.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/time.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/acos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/atan.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/ceil.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/copysign.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/cos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/exp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fdim.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/floor.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmax.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmod.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fpclassify.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/frexp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/modf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__cosdf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzero.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzerof.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalid.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidl.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2_large.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__sindf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/acosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/asinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/atanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/cosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/erff.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expm1f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/log1pf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/sinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/tanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/nextafter.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/pow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remainder.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remquo.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/rint.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/round.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sqrt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/trunc.c --64 --floatbv --output /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64_fp.goto
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c:150:41: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __ESBMC_thread_key_destructors[i](l->value);
                                        ^~~~~~~~
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c:20:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
}
^
Segmentation fault
[6/21] Generating libc model /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32_fp.goto
FAILED: src/c2goto/clib32_fp.goto /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32_fp.goto
cd /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto && /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/c2goto /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/builtin_libs.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/ctype.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/fenv.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/getopt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/inet.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/io.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/kernel.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/semaphore_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/stdlib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/string.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/syserr.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/time.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/acos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/atan.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/ceil.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/copysign.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/cos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/exp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fdim.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/floor.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmax.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmod.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fpclassify.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/frexp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/modf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__cosdf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzero.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzerof.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalid.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidl.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2_large.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__sindf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/acosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/asinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/atanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/cosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/erff.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expm1f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/log1pf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/sinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/tanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/nextafter.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/pow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remainder.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remquo.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/rint.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/round.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sqrt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/trunc.c --32 --floatbv --output /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32_fp.goto
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c:150:41: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __ESBMC_thread_key_destructors[i](l->value);
                                        ^~~~~~~~
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c:20:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
}
^
Segmentation fault
[7/21] Generating libc model /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32.goto
FAILED: src/c2goto/clib32.goto /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32.goto
cd /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto && /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/c2goto /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/builtin_libs.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/ctype.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/fenv.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/getopt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/inet.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/io.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/kernel.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/semaphore_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/stdlib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/string.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/syserr.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/time.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/acos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/atan.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/ceil.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/copysign.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/cos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/exp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fdim.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/floor.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmax.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmod.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fpclassify.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/frexp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/modf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__cosdf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzero.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzerof.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalid.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidl.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2_large.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__sindf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/acosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/asinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/atanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/cosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/erff.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expm1f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/log1pf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/sinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/tanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/nextafter.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/pow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remainder.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remquo.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/rint.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/round.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sqrt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/trunc.c --32 --fixedbv -D__ESBMC_FIXEDBV --output /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib32.goto
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c:150:41: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __ESBMC_thread_key_destructors[i](l->value);
                                        ^~~~~~~~
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c:20:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
}
^
Segmentation fault
[8/21] Generating libc model /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64.goto
FAILED: src/c2goto/clib64.goto /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64.goto
cd /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto && /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/c2goto /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/builtin_libs.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/ctype.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/fenv.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/getopt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/inet.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/io.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/kernel.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/semaphore_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/stdlib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/string.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/syserr.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/time.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/acos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/atan.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/ceil.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/copysign.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/cos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/exp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fdim.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/floor.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmax.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmod.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fpclassify.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/frexp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/modf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__cosdf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzero.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzerof.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalid.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidl.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2_large.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__sindf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/acosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/asinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/atanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/cosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/erff.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expm1f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/log1pf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/sinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/tanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/nextafter.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/pow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remainder.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remquo.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/rint.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/round.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sqrt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/trunc.c --64 --fixedbv -D__ESBMC_FIXEDBV --output /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64.goto
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c:150:41: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __ESBMC_thread_key_destructors[i](l->value);
                                        ^~~~~~~~
/home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c:20:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
}
^
Segmentation fault
ninja: build stopped: subcommand failed.
```

Reply:
We should understand the issues before submitting the PRs, otherwise we may introduce behaviours in the code base that we don't expect.

I suggest you run c2goto on a debugger to understand why it's crashing.

Reply:
Hi @mikhailramalho ,

Sorry I don't know how to debug `c2goto` yet, Could you please give me some instructions?

As for this error, I think I've found the problem, according to Clang's documentation, the type returned by `EnumType::desugar()` should be `clang::EnumType`. So we may have an infinite recursion when we call `get_type()`.

For this example:
```
enum Color {
    RED,
    GREEN,
    BLUE
};
```

> For the `desugar()` method of `clang::EnumType`, it usually returns the enumeration type itself, because enumeration types are not syntactic sugar reducts from other types. It is a primitive type that represents a fixed set of named constants. For this reason, the `desugar()` method will usually not cast to another type when working with `clang::EnumType`; It returns a representation of the enumerated type itself.

Reply:
You can just run the failing c2goto command under gdb, e.g.
```
gdb --args /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/c2goto /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/builtin_libs.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/ctype.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/fenv.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/getopt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/inet.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/io.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/kernel.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/pthread_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/semaphore_lib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/setjmp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/stdlib.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/string.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/syserr.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/time.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/acos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/atan.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/ceil.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/copysign.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/cos.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/exp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fdim.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/floor.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmax.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fmod.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/fpclassify.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/frexp.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/modf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__cosdf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzero.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_divzerof.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalid.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_invalidl.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_oflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_uflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__math_xflowf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2_large.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__rem_pio2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/__sindf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/acosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/asinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/atanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/cosf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/erff.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/exp2f_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/expm1f.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/log1pf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/logf_data.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/sinf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/musl/tanhf.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/nextafter.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/pow.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remainder.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/remquo.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/rint.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/round.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sin.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/sqrt.c /home/lxzy/ESBMC_Project/esbmc/src/c2goto/library/libm/trunc.c --64 --floatbv --output /home/lxzy/ESBMC_Project/esbmc/build/src/c2goto/clib64_fp.goto
```

Reply:
hum... that probably means that the enum type is already desugared, can you check clang's codebase and try to find how the underlying type of the enum is retrieved? I imagine it should be something similar to how it's done in the link I've sent; maybe that's even the solution but I would double-check it first.